mindmeld/auto_annotator.py

```diff
@@ -835,5 +839,300 @@ def valid_entity_check(self, entity):
+        )
+        en_sentence = self.translator.translate(sentence, target_language="en")
+        en_entities = self.en_annotator.parse(en_sentence)
+
+        selected_candidates = []
+        for entity in en_entities:
+            i = 0
+            while i < len(candidates):
+                if entity["dim"] == candidates[i]["entity_type"]:
+                    if entity["value"] == candidates[i]["value"]:
+                        selected_candidates.append(candidates[i])
+                        break
+                    if (
+                        self.translator.translate(
+                            entity["body"], target_language=language
+                        )
+                        == candidates[i]["body"]
+                    ):
+                        selected_candidates.append(candidates[i])
+                        break
+                i += 1
+        return selected_candidates
+
+    def _parse_without_translator(self, sentence, language=None):
+        """Parse helper function to be used if a translator is not present.
+
+        Args:
+            sentence (str): Sentence to detect entities.
+            language (str): Language code for the language of the given sentence
+        Returns: entities (list): List of duckling candidates.
+        """
+        candidates = self.en_annotator.duckling.get_candidates_for_text(
+            sentence, language=language
+        )
+        final_spans = MultiLingualAnnotator._get_final_spans(candidates, sentence)
+        selected_candidates = []
+        for span in final_spans:
+            for candidate in candidates:
+                if span == (candidate["start"], candidate["end"]):
+                    if not (
+                        candidate["dim"] == "amount-of-money"
+                        and candidate["value"]["unit"] == "unknown"
+                    ):
+                        selected_candidates.append(candidate)
+                        break
+        return selected_candidates
+
+    @staticmethod
+    def convert_duckling_candidates(duckling_candidates):
+        """Converts duckling candidates into entity dictionaries.
+        Args:
+            duckling_candidates (list): List of Duckling candidates
+        Returns:
+            entities (list): List of converted entities.
+        """
+        return [
+            {
+                "body": entity["body"],
+                "start": entity["start"],
+                "end": entity["end"],
+                "dim": entity["entity_type"],
+                "value": entity["value"],
+            }
+            for entity in duckling_candidates
+        ]
+
+    @property
+    def supported_entity_types(self):  # pylint: disable=W0236
+        """
+        Returns:
+            supported_entity_types (list): List of supported entity types.
+        """
+        return SPACY_ANNOTATOR_SUPPORTED_ENTITIES
+
+    @staticmethod
+    def _has_overlap(span_one, span_two):
```

Comment:
I think the `_has_overlap `, `_is_larger ` should be part of the `Span` class. You can implement the primitive functions for `<` ops there.

Reply:
Added `__gt__, __ge__, __lt__, __le__`